testpar/API/H5_api_async_test_parallel.c

```diff
@@ -3567,107 +3593,126 @@ test_file_reopen(void)
     }
     H5E_END_TRY
 
-    return 1;
+    return;
 }
 
 /*
  * Cleanup temporary test files
  */
 static void
-cleanup_files(void)
+test_async_file_cleanup(const void H5_ATTR_UNUSED *params)
 {
     char file_name[64];
     int  i;
 
     if (MAINPROCESS) {
-        H5Fdelete(PAR_ASYNC_API_TEST_FILE, H5P_DEFAULT);
+        remove_test_file(NULL, PAR_ASYNC_API_TEST_FILE);
```

Comment:
Creation of this file is skipped based on VOL cap flags, so cleanup probably should be too

Reply:
True, though file cleanup should assume that some files may be missing anyway, so it's simpler and less code just to always try deleting things.